src/checkbox/internal.tsx

```diff
@@ -39,6 +38,7 @@ const InternalCheckbox = React.forwardRef<CheckboxProps.Ref, InternalProps>(
     },
     ref
   ) => {
+    const { ariaDescribedby, ariaLabelledby } = useFormFieldContext(rest);
```

Comment:
Why using the context here and not inside AbstractSwitch?

If you do it inside the abstract switch it would also be possible to use the form-field controlId:

```
const formFieldContext = useFormFieldContext(rest);
const selfControlId = useUniqueId();
const controlId = propsControlId || formFieldContext.controlId || selfControlId;
```

Reply:
Inside AbstractSwitch may not work for RadioGroup -- we want to add ariaDescribedby and ariaLabelledby to RadioGroup but AbstractSwitch is for individual RadioButton.

Reply:
Can we then make the `controlId` property of the abstract switch required and compute this outside to still have the precedence `propsControlId || formFieldContext.controlId || selfControlId `? (for the radio group it is going to be selfControlId always).

Reply:
It is the same for controlId(miss it in the last reply), AbstractSwitch controlId is for the individual RadioButton `controlId={item.controlId}`  not for the RadioGroup.

Reply:
Yes, so for the checkbox we can do:

```
function Checkbox(props) {
  const context = useFormFieldContext(props);
  const selfControlId = useUniqueId();
  const controlId = props.controlId || context.controlId || selfControlId;
  return <AbstractSwitch
    controlId={controlId}
    ariaLabelledby={context.ariaLabelledby}
    ariaDescribedby={context.ariaDescribedby}
    ...
  />
}
```

and for radio group:

```
function RadioGroup(props) {
  const context = useFormFieldContext(props);
  const selfControlId = useUniqueId();
  const controlId = props.controlId || context.controlId || selfControlId;
  return <div role="radio-group"
    ariaLabelledby={context.ariaLabelledby}
    ariaDescribedby={context.ariaDescribedby}
    ...
  >
    {items.map((...) => <RadioButton {...} />)}
  </div>;
}

function RadioButton(props) {
  const controlId = useUniqueId();
  return <AbstractSwitch controlId={controlId} ... />;
}
```

Reply:
Offline discussion: the control-id is not passed from form-field to the abstract-switch to avoid activating checkbox/radio/toggle when clicking on the form-field label. The current implementation addresses that.